### PR TITLE
Feat/common

### PR DESCRIPTION
**Pull request by Ryanmtate**
Rebase #30 first.

The `common.rs` module contains common types that will be used by several different interfaces within the sdk. For example, it includes uniffi conversions of common types, like `Url` and `Uuid` and moves the `Key` and `Value` types from the storage manager, so they can be re-used.

As we identify more types that make sense to go into common, e.g. `ClaimFormatDesignation`, etc., this would be the logical place to migrate those types.

### CODE REVIEW

**Comment by Ryanmtate** [comment]
NOTE: This branch is based on #30, change the branch to main once #30 is merged.

File: src/storage_manager.rs [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 
-uniffi::custom_newtype!(Key, String);
-pub struct Key(pub String);
+// uniffi::custom_newtype!(Key, String);
+// pub struct Key(pub String);
 
-uniffi::custom_newtype!(Value, Vec<u8>);
-pub struct Value(pub Vec<u8>);
+// uniffi::custom_newtype!(Value, Vec<u8>);
+// pub struct Value(pub Vec<u8>);
```
Comment (Ryanmtate) [inline src/storage_manager.rs:9]: Remove dead code.

File: Cargo.toml [hunk 2]
```diff
@@ -38,3 +38,10 @@ base64 = "0.22.0"
 
 [build-dependencies]
 uniffi = { version = "0.28.1", features = ["build"] }
+
+[patch.crates-io]
+isomdl = { git = "https://github.com/spruceid/isomdl", rev = "1f4f762" }
+
+[patch.'https://github.com/spruceid/oid4vp-rs']
+# TODO: replace this once oid4vp is merged.
+oid4vp = { git = "https://github.com/Ryanmtate/oid4vp-rs", rev = "2e453ca6" }
```
Comment (Ryanmtate) [inline Cargo.toml:47]: NOTE: Check if these revisions need to be bumped.

Comment (Ryanmtate) [reply Cargo.toml]: bumped the `oid4vp` revision.

**Comment by rschulman** [comment]
LGTM

**Review by rschulman — APPROVED** [review]
I should have done this in "Review changes" mode... LGTM.

**Review by rschulman — APPROVED** [review]
One small note about `local.properties` but really this is fine and good to go.

File: README.md [hunk 3]
```diff
@@ -38,9 +38,21 @@ which is a manually triggered action.
...
 ```
 
+If you get this error:
+```
+> java.io.FileNotFoundException: .../local.properties (No such file or directory)
+```
+
+run:
+```bash
+touch local.properties
```
[the remark this answers is not in the document]
Comment (cobward) [reply README.md:51]: Yeah for me I just needed to create the file. We cna update later with further instructions if needed.

**Pull request merged**